docs/gateway/create-a-prompt-template.mdx

```diff
@@ -0,0 +1,263 @@
+                    "arguments": {"topic": "artificial intelligence"},
+                }
+            ],
+        },
+    ],
+)
+```
+
+</Tab>
+
+<Tab title="HTTP">
+
+Use your template during inference by sending a `template` content block with the template name and arguments.
+
+```bash
+curl -X POST http://localhost:3000/inference \
+  -H "Content-Type: application/json" \
+  -d '{
+    "function_name": "fun_fact",
+    "input": {
+      "messages": [
+        {
+          "role": "user",
+          "content": [
+            {
+              "type": "template",
+              "name": "fun_fact_topic",
+              "arguments": {
+                "topic": "artificial intelligence"
+              }
+            }
+          ]
+        }
+      ]
+    }
+  }'
+```
+
+</Tab>
+
+</Tabs>
+
+</Step>
+
+</Steps>
+
+## Set up a template schema
+
+When you have multiple variants for a function, it becomes challenging to ensure all templates use consistent variable names and types.
+Schemas solve this by defining a contract that validates template variables and catches configuration errors before they reach production.
+Defining a schema is optional but recommended.
+
+<Steps>
+
+<Step title="Create a schema">
+
+Create a [JSON Schema](https://json-schema.org/) for the variables used by your templates.
+
+Let's define a schema for our previous example, which includes only a single variable `topic`:
+
+```json title="config/functions/fun_fact/fun_fact_topic_schema.json"
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "topic": {
+      "type": "string"
+    }
+  },
+  "required": ["topic"],
+  "additionalProperties": false
+}
+```
+
+<Tip>
+
+LLMs are great at generating JSON Schemas.
+For example, the schema above was generated with the following request:
+
+```txt
+Generate a JSON schema with a single field: `topic`.
+The `topic` field is required. No additional fields are allowed.
+```
+
+You can also export JSON Schemas from [Pydantic models](https://docs.pydantic.dev/latest/concepts/json_schema/) and [Zod schemas](https://www.npmjs.com/package/zod-to-json-schema).
+
+</Tip>
+
+</Step>
+
+<Step title="Configure a schema">
+
+Then, declare your schema in your function definition using `schemas.your_schema_name.path`.
+This will ensure that every variant for the function has a template named `your_schema_name`.
+
+In our example above, this would mean updating the function definition to:
+
+```toml
+[functions.fun_fact]
+type = "chat"
+schemas.fun_fact_topic.path = "functions/fun_fact/fun_fact_topic_schema.json" # relative to this file // [!code ++]
+
+[functions.fun_fact.variants.gpt_5_mini]
+type = "chat_completion"
+model = "openai::gpt-5-mini"
+templates.fun_fact_topic.path = "functions/fun_fact/gpt_5_mini/fun_fact_topic_template.minijinja" # relative to this file
+```
+
+</Step>
+
+</Steps>
+
+## Re-use prompt snippets
+
+You can enable template file system access to reuse shared snippets.
```

Comment:
can we actually include an example of the include tag and the link to the minijinja docs for it? this is an FAQ